c10/core/TensorImpl.h

```diff
@@ -1404,6 +1448,17 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
   // and its subclasses to find which fields are copied by value.
   bool allow_tensor_metadata_change_ = true;
 
+  // An opaque handle `opaque_handle_` allows customized data layout other than the default
```

Comment:
move this block above the bool variables. It'd allow you to increate TensorImpl size by only 1 word instead of 2 words like now

Reply:
> move this block above the bool variables. It'd allow you to increate TensorImpl size by only 1 word instead of 2 words like now

@dzhulgakov Above what bool variable? I couldn't figure out the right place to add to save 64-bit.

Reply:
Above all bool variables, aka. put this `opaque_handle_` field to be above line 1436 `bool is_contiguous_ = true;`. We have 5 bools, right now it's the first 4 get padded to 1 word, opaque_handle_ takes 1 word and the 5th bool takes another 1 word. If you put opaque_handle_ at the beginning, then opaque_handle_ takes 1 word and all the other 5 bools will together take 1 word (on 64-bit system).